iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientConfiguration.java

```diff
@@ -208,8 +195,9 @@ private void configSasAuth(IotHubConnectionString iotHubConnectionString) {
 
     ClientConfiguration(IotHubConnectionString iotHubConnectionString, IotHubClientProtocol protocol, ClientOptions clientOptions)
     {
-        String mqttGatewayHostName = iotHubConnectionString.getMqttGatewayHostName();
-        if (mqttGatewayHostName != null && !mqttGatewayHostName.isEmpty())
+        String gatewayHostName = iotHubConnectionString.getGatewayHostName();
+        GatewayType gatewayType = clientOptions.getGatewayType();
+        if (gatewayHostName != null && !gatewayHostName.isEmpty() && gatewayType == GatewayType.E4K)
```

Comment:
Should we throw an exception of the gatewayType is set to E4K, but the hostname isn't supplied?

Reply:
Good point!